src/topography.f90

```diff
@@ -72,6 +75,19 @@ type(topography_t) function topography_constructor(filename) result(topog)
     call handle_error(nf90_get_att(ncid, depth_id, 'maximum_depth', topog%max_depth), isfatal=.false.)
     call handle_error(nf90_get_att(ncid, depth_id, 'nonadvective_cells_removed', topog%nonadvective_cells_removed), isfatal=.false.)
 
+    if (present(grid_type)) then
+      topog%grid_type = grid_type ! grid_type arg overrides value in file
```

Comment:
Wouldn't it be better to check if the value given in the command line is consistent with the value on the file? Otherwise, if I understand correctly, it's possible to change the grid type on file when running `deseas`, which is a bit weird.

Reply:
This is an intentional feature, otherwise there's no way to alter the grid type on file (which otherwise defaults to B)

Reply:
I'll add some words to that effect in the docs.

Reply:
The main purpose of grid_type in the file is to indicate to fill_fraction, fix_nonadvective and check_nonadvective what connectivity rules to use when checking whether the number of seas has changed, ie to indicate the rules used previously by deseas when lakes_removed was set to 'yes'.

Reply:
Ah, I see. Yes, this definitely needs to be documented, both in the docs and in the code.

Reply:
ok done - let me know if you're happy with that

Reply:
Thanks, much better now. Might be good to have a more comprehensive discussion about default values and how to handle missing attributes. We can do that in #42 I think.